tests/cc/test_looping.cq

```diff
@@ -0,0 +1,222 @@
+# File:         test_looping.cq
+# Purpose:      test cQasm looping with CC backend
+# Notes:        most constructs make no practical sense, we're just checking the constructs and code generation
+
+version 1.2
+qubits 17
+
+pragma @ql.name("test_looping")
+pragma @ql.platform("config_cc_s17_direct_iq_openql_0_10.json")
+
+.def
+map qInput1 = q[0]
+map qInput2 = q[1]
+map qAncilla3 = q[2]
+map qOutput4 = q[3]
+
+# var mAncilla3: bool   encountered incompatible object reference to mAncilla3 (size=0)
+map mAncilla3 = b[2]
+
+map i = creg(0)
+map j = creg(1)
+
+.init
+set i = 0;
+
+.forLoop1
+for (i=0; i<10000; i=i+1) {
+#    if (!mAncilla3) { # Error: in expression '!(bit)q[2]': index 1 is out of range, size is 1
+#    if (!b[2]) {
+#        break
+#    }
+}
+
+
+.forLoop2
+for (; i<10000; )  {
+}
+
+.forLoop3
+for (; true; )  {  # NB: condition cannot be empty, defined as: utils::One<ir::Expression>
+}
+
+.forLoopConstProp
+for (; i<5*5; )  {
+}
+
+
+.forLoopTwice  # must generate unique labels
+for (; true; )  {
+}
+for (; true; )  {
+}
+
+.forLoopNested
+for (; true; )  {
+    for (; true; )  {
+    }
+}
+
+.forLoopBreak
+for (; true; )  {
+    measure_fb q[0]
+    if (b[0]) {
+        break;
+    }
+}
+
+.forLoopBreak2
+for (; true; )  {
+    if(i==1) {
+        break;
+    }
+}
+
+.forLoopBreak3
+for (; true; )  {
+    if(i==1) {
+        if(j==1) {
+            break;
+        }
+    }
+}
+
+.forLoopContinue
+for (; true; )  {
+    if (!b[0]) { # NB: inverted condition
+        continue;
+    }
+}
+
+.foreachLoopUp
+foreach (i = 0..10) {
+}
+
+
+.foreachLoopDown0 # counting down towards zero gets special treatment
+foreach (i = 10..0) {
+}
+
+
+.foreachLoopDown
+foreach (i = 10..5) {
+}
+
+
+# FIXME: we do not handle negative numbers
+#.foreachLoopDown2
+#foreach (i = 10..-5) {
+#}
+
+
+.whileLoop
+while (i<10000) {
+}
+
+
+.repeatUntilLoop
+repeat {
+} until (i==0);
+
+
+.ifStatement
+if (i==0) {
+}
+
+.ifElseStatement
+if (i==0) {
+} else {
+}
+
+.ifElifElseStatement
+if (i==0) {
+} else if (i==1)  {
+} else {
+}
+
+
+
+.condLt
+while (i<10000) {
+}
+
+.condGt
+while (i>10000) {
+}
+
+.condEq
+while (i==10000) {
+}
+
+.condNe
+while (i!=10000) {
+}
+
+.condTrue
+while (true) {
+}
+
+.condFalse
+while (false) {
+}
+
+#.condBit
+#while (mAncilla3) { # Error: encountered incompatible object reference to mAncilla3 (size=0)
+#}
+
+.condBit2
+while (b[0]) {
+}
+
+#.condBit3
+#while (b[0:2]) { # Error: Usage error: in pass reader, phase main: unexpected number of single-gate-multiple-qubit bit indices specified; found 3, expected 1
```

Comment:
See above, so sgmq handling in pass reader is not right.
What is the problem?

Reply:
Not sure, see above

Reply:
I suggest to add to investigate this in the same issue as just created (on cond(b[...]) x q[...]). It is similar.

Reply:
Added it to https://github.com/QuTech-Delft/OpenQL/issues/446